pytket/extensions/qiskit/qiskit_convert.py

```diff
@@ -245,6 +246,36 @@ def _qpo_from_peg(peg: PauliEvolutionGate, qubits: List[Qubit]) -> QubitPauliOpe
     return QubitPauliOperator(qpodict)
 
 
+def _string_to_circuit(
+    circuit_string: str, n_qubits: int, qiskit_instruction: Instruction
+) -> Circuit:
+    """Helper function to handle strings in QuantumCircuit.initialize
+    and QuantumCircuit.prepare_state"""
+
+    circ = Circuit(n_qubits)
+    # Check if Instruction is Initialize or Statepreparation
+    # If Initialize, add resets
+    if isinstance(qiskit_instruction, Initialize):
+        for qubit in circ.qubits:
+            circ.add_gate(OpType.Reset, [qubit])
+
+    # We iterate through the string in reverse to add the
+    # gates in the correct order (endian-ness).
+    for count, char in enumerate(reversed(circuit_string)):
+        if char == "0":
+            pass
+        if char in ("1", "-"):
+            circ.X(count)
+        if char in ("+", "-", "r", "l"):
+            circ.H(count)
+            if char == "r":
+                circ.S(count)
+            elif char == "l":
+                circ.Sdg(count)
```

Comment:
This is quite confusing; I think it would be easier to follow if we had single-option cases, with `elif`:

```python
if char == "0":
    pass
elif char == "1":
   circ.X(count)
elif ...

Reply:
Okay will separate the cases if thats preferred.

Reply:
Done